lib/components/viewers/amenities-panel.js

```diff
@@ -0,0 +1,192 @@
+import { getCompanyIcon } from '@opentripplanner/icons/lib/companies'
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import styled from 'styled-components'
+
+import { ComponentContext } from '../../util/contexts'
+
+import RelatedPanel from './related-panel'
+
+const StyledParkAndRideIcon = styled.div`
+  background: #000;
+  border-radius: 17px;
+  color: #fff;
+  font-size: 16px;
+  font-weight: bold;
+  height: 22px;
+  line-height: 0px;
+  margin-right: 5px;
+  padding-left: 7px;
+  padding-top: 12px;
+  width: 22px;
+`
+
+const parkAndRideMarker = <StyledParkAndRideIcon>P</StyledParkAndRideIcon>
+
+class AmenitiesPanel extends Component {
```

Comment:
This component and all of the methods within could be broken out into functional components that used `useState` and `useContext` as needed.